host/src/exit.rs

```diff
@@ -0,0 +1,14 @@
+#![allow(unused_variables)]
+
+use crate::{wasi_exit, WasiCtx};
+
+#[async_trait::async_trait]
+impl wasi_exit::WasiExit for WasiCtx {
+    async fn exit(&mut self, status: Result<(), ()>) -> anyhow::Result<()> {
+        let status = match status {
+            Ok(()) => 0,
+            Err(()) => 1,
+        };
+        std::process::exit(status)
```

Comment:
Process exit is inappropriate - instead, we need to return a trap here. We shoul